src/test/java/org/reaktivity/specification/socks/internal/SocksFunctionsTest.java

```diff
@@ -55,22 +59,48 @@ public void shouldLoadFunctions() throws Exception
     @Test
     public void shouldBuildRouteEx() throws Exception
     {
-        byte[] bytes = SocksFunctions.routeEx("example.com", 8080);
+        byte[] bytes = SocksFunctions.routeEx().address("example.com").port(8080).build();
         DirectBuffer buffer = new UnsafeBuffer(bytes);
         SocksRouteExFW routeEx = new SocksRouteExFW().wrap(buffer, 0, buffer.capacity());
 
         assertEquals("example.com", routeEx.address().asString());
         assertEquals(8080, routeEx.port());
     }
 
+
     @Test
     public void shouldBuildBeginEx() throws Exception
     {
-        byte[] bytes = SocksFunctions.beginEx("example.com", 8080);
+        byte[] bytes = SocksFunctions.beginEx().typeId(0).address("example.com").port(8080).build();
         DirectBuffer buffer = new UnsafeBuffer(bytes);
         SocksBeginExFW beginEx = new SocksBeginExFW().wrap(buffer, 0, buffer.capacity());
 
         assertEquals("example.com", beginEx.address().asString());
         assertEquals(8080, beginEx.port());
     }
+
+    @Test
+    public void shouldEncodeSocksEndExAsUnsubscribe()
+    {
+        final byte[] array = SocksFunctions.endEx()
+                .typeId(0)
+                .build();
+
+        DirectBuffer buffer = new UnsafeBuffer(array);
+        SocksEndExFW socksEndEx = new SocksEndExFW().wrap(buffer, 0, buffer.capacity());
+        assertEquals(0, socksEndEx.typeId());
+    }
+    @Test
+    public void shouldEncodeSocksAbortExAsUnsubscribe()
+    {
+        final byte[] array = SocksFunctions.abortEx()
+                .typeId(0)
+                .reason(0xf9)
+                .build();
+
+        DirectBuffer buffer = new UnsafeBuffer(array);
+        SocksAbortExFW socksAbortEx = new SocksAbortExFW().wrap(buffer, 0, buffer.capacity());
+        assertEquals(0, socksAbortEx.typeId());
+        assertEquals(0xf9, socksAbortEx.reason());
+    }
```

Comment:
As mentioned above, these last 2 methods can be removed, as we only need `SOCKS` stream frame extensions for `SocksBeginEx`.